### DISCUSSION TITLE
Custom Template for code gen not working

### DISCUSSION THREAD

**Discussion by reach2rv** [serenity-is/Serenity#7219] [Help] [answered]
Hi,

I am using custom template for row server typing 

```
export interface {{RowClassName}} {%{{}%}{{for f in Fields}}
    {{f.PropertyName}}?: {{f.TSType}};{{end}}{{for j in Joins}}{{for k in j.Fields}}
    {{k.PropertyName}}?: {{k.TSType}};{{end}}{{end}}
}

export abstract class {{RowClassName}} {%{{}%}{{if Identity}}
    static readonly idProperty = '{{Identity}}';{{end}}{{if NameField}}
    static readonly nameProperty = '{{NameField}}';{{end}}
    static readonly localTextPrefix = '{{ModuleDot}}{{ClassName}}';

    static readonly deletePermission = '{{Permission}}';
    static readonly insertPermission = '{{Permission}}';
    static readonly readPermission = '{{Permission}}';
    static readonly updatePermission = '{{Permission}}';
    static readonly postingPermission = '{{Permission}}';

    static readonly Fields = {{QIMPORT "fieldsProxy"}}<{{RowClassName}}>();
}
```

and have enabled custom template in sergen.json with

`  "CustomTemplates": "Modules\\Common\\CodeGenTemplates",`

but servertypings for row are still generated with default templates. am I missing something?

**Comment by reach2rv** [comment]
update I have added those permissions on code generator it works if I use dotnet sergen s but if I rebuild it removes posting permission.

**Reply by reach2rv** [reply]
@VictorTomaili - apologies for tagging you. But I have tried all possible solutions without any success. can you please shed some light on this. 

PS: does it has anything to do with pro coder?

**Reply by VictorTomaili** [reply]
Custom templates are not supported by coder pro

**Reply by reach2rv** [reply]
Thank you. Is there is any way to disable server typing generation by coder pro?

**Reply by VictorTomaili** [reply, marked as answer]
You can remove it completely and use sergen instead

**Reply by reach2rv** [reply]
Found a solution @VictorTomaili @volkanceylan can I post it here?

**Reply by VictorTomaili** [reply]
Related pro packages, you need to use StartSharp repository. Don't post pro packages related content in community area.
